nsxt/resource_nsxt_policy_ipsec_vpn_service.go

```diff
@@ -0,0 +1,242 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_1s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var IPSecVpnServiceIkeLogLevelTypes = []string{
+	model.IPSecVpnService_IKE_LOG_LEVEL_DEBUG,
+	model.IPSecVpnService_IKE_LOG_LEVEL_INFO,
+	model.IPSecVpnService_IKE_LOG_LEVEL_WARN,
+	model.IPSecVpnService_IKE_LOG_LEVEL_ERROR,
+	model.IPSecVpnService_IKE_LOG_LEVEL_EMERGENCY,
+}
+
+func resourceNsxtPolicyIPSecVpnService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyIPSecVpnServiceCreate,
+		Read:   resourceNsxtPolicyIPSecVpnServiceRead,
+		Update: resourceNsxtPolicyIPSecVpnServiceUpdate,
+		Delete: resourceNsxtPolicyIPSecVpnServiceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"gateway_path": getPolicyGatewayPathSchema(),
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Enable/Disable IPSec VPN service.",
+				Optional:    true,
+				Default:     true,
+			},
+			"haSync": {
```

Comment:
please use snake case naming convention here and below - `ha_sync`, `bypass_rule`..